standard/patterns.md

```diff
@@ -140,6 +144,157 @@ Given a pattern input value ([§11.1](patterns.md#111-general)) *e*, if *designa
 
 It is an error if the name `var` would bind to a type where a *var_pattern* is used.
 
+§type-pattern-new-clause Type pattern
+
+A *type_pattern* is used to test that the pattern input value ([§11.1](patterns.md#111-general)) has a given type.
+
+```ANTLR
+type_pattern
+    : type
+    ;
+```
+
+The runtime type of the value is tested against *type* using the same rules specified in the is-type operator ([§12.12.12.1](expressions.md#1212121-the-is-type-operator)). If the test succeeds, the pattern matches that value. It is a compile-time error if the *type* is a nullable type. This pattern form never matches a `null` value.
+
+§relational-pattern-new-clause Relational pattern
+
+A *relational_pattern* is used to relationally test the pattern input value ([§11.1](patterns.md#111-general)) against a constant value.
+
+```ANTLR
+relational_pattern
+    : '<'  constant_expression
+    | '<=' constant_expression
+    | '>'  constant_expression
+    | '>=' constant_expression
+    ;
+```
+
+Relational patterns support the relational operators `<`, `<=`, `>`, and `>=` on all of the built-in types that support such binary relational operators with both operands having the same type: `sbyte`, `byte`, `short`, `ushort`, `int`, `uint`, `long`, `ulong`, `char`, `float`, `double`, `decimal`, `nint`, `nuint`, and enums.
+
+It is a compile-time error if `constant_expression`is `double.NaN`, `float.NaN`, or `null_literal`.
+
+When the input value has a type for which a suitable built-in binary relational operator is defined, the evaluation of that operator is taken as the meaning of the relational pattern.  Otherwise, the input value is converted to the type of `constant_expression` using an explicit nullable or unboxing conversion.  It is a compile-time error if no such conversion exists.  The pattern is considered to not match if the conversion fails.  If the conversion succeeds, the result of the pattern-matching operation is the result of evaluating the expression `e «op» v` where `e` is the converted input, «op» is the relational operator, and `v` is the `constant_expression`.
+
+> *Example*:
+>
+> <!-- Example: {template:"standalone-console", name:"RelationalPattern1", inferOutput:true} -->
+> ```csharp
+> Console.WriteLine(Classify(13));
+> Console.WriteLine(Classify(double.NaN));
+> Console.WriteLine(Classify(2.4));
+>
+> static string Classify(double measurement) => measurement switch
+> {
+>     < -4.0 => "Too low",
+>     > 10.0 => "Too high",
+>     double.NaN => "Unknown",
+>     _ => "Acceptable",
+> };
+> ```
+>
+> The output produced is
+>
+> ```console
+> Too high
+> Unknown
+> Acceptable
+> ```
+>
+> *end example*
+
+§logical-pattern-new-clause Logical pattern
+
+A *logical_pattern* is used to negate a pattern input value ([§11.1](patterns.md#111-general)) or to combine that value with a pattern using a Boolean operator.
```

Comment:
No, it doesn't negate the input value. It inverts (in the boolean sense) the result of the pattern-match. Similarly, the combination doesn't happen with the value, but with the result of pattern-match tests.

Reply:
If is an operation of many names: not, invert & negate, and in C# it is indeed called logical negation (§12.9.4).

Reply:
I'm not concerned with which of those names we use. I'm pointing out that the meaning is not "to negate a pattern input value", but rather to negate the result of determining whether or not the input value matches the pattern.